File: tests/py/testPolymorph.py
import swigex as mf

class ChildTwo(mf.ParentBase):
    def __init__(self):
        super(ChildTwo,self).__init__()
    
    def getHello(self):
        return "ChildTwo in Python - Hello"

pa = mf.ParentBase()
mf.showHello(pa)

co = mf.ChildOne()
mf.showHello(co)
    
ct = ChildTwo()
mf.showHello(ct)

File: tests/r/testPolymorph.r
suppressWarnings(suppressMessages(library(swigex)))

pa = ParentBase()
invisible(showHello(pa))

co = ChildOne()
invisible(showHello(co))

# TODO : Cannot inherits from ParentBase in R
#ct = ParentBase()
#class(ct) <- c("ChildTwo", "ParentBase")
#getHello.ChildTwo <- function(obj)
#{
#  return("ChildTwo - Hello")
#}
#
#showHello(ct)
